dataclients/kubernetes/kube.go

```diff
@@ -595,3 +595,19 @@ func compareStringList(a, b []string) []string {
 	}
 	return c
 }
+
+// addTLSCertToRegistry adds a TLS certificate to the certificate registry per host using the provided
+// Kubernetes TLS secret
+func addTLSCertToRegistry(cr *certregistry.CertRegistry, logger *logger, hosts []string, secret *secret) {
+	cert, err := generateTLSCertFromSecret(secret)
+	if err != nil {
+		logger.Errorf("Failed to generate TLS certificate from secret: %v", err)
```

Comment:
Maybe pass secretId and log it as well

Reply:
Even though the error from generateTLSCertFromSecret is pretty explicit about the reason?

Reply:
Logger logs namespace and ingress/routegroup so I guess its enough context to debug.